controllers/stargate/vector.go

```diff
@@ -0,0 +1,159 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *StargateReconciler) reconcileVectorConfigMap(
+	ctx context.Context,
+	stargate api.Stargate,
+	actualDc *cassdcapi.CassandraDatacenter,
+	remoteClient client.Client,
+	dcLogger logr.Logger,
+) (ctrl.Result, error) {
+	namespace := stargate.Namespace
+	configMapKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      stargatepkg.VectorAgentConfigMapName(actualDc.Spec.ClusterName, actualDc.Name),
+	}
+	if stargate.Spec.Telemetry.IsVectorEnabled() {
+		// Create the vector toml config content
+		toml, err := CreateVectorToml(stargate.Spec.Telemetry)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		desiredVectorConfigMap := stargatepkg.CreateVectorConfigMap(namespace, toml, *actualDc)
+		annotations.AddHashAnnotation(desiredVectorConfigMap)
+
+		// Check if the vector config map already exists
+		actualVectorConfigMap := &corev1.ConfigMap{}
+
+		if err := remoteClient.Get(ctx, configMapKey, actualVectorConfigMap); err != nil {
+			if errors.IsNotFound(err) {
+				if err := ctrl.SetControllerReference(&stargate, desiredVectorConfigMap, r.Scheme); err != nil {
+					dcLogger.Error(err, "Failed to set controller reference on new Stargate Vector ConfigMap", "ConfigMap", configMapKey)
+					return ctrl.Result{}, err
+				} else if err := remoteClient.Create(ctx, desiredVectorConfigMap); err != nil {
+					if errors.IsAlreadyExists(err) {
+						// the read from the local cache didn't catch that the resource was created
+						// already; simply requeue until the cache is up-to-date
+						dcLogger.Info("Stargate Vector Agent configuration already exists, requeueing")
+						return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+					} else {
+						dcLogger.Error(err, "Failed to create Stargate Vector Agent ConfigMap")
+						return ctrl.Result{}, err
+					}
+				}
+				// Requeue to ensure the config map can be retrieved
+				return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+			} else {
+				dcLogger.Error(err, "Failed to get Stargate Vector Agent ConfigMap")
+				return ctrl.Result{}, err
+			}
+		}
+
+		actualVectorConfigMap = actualVectorConfigMap.DeepCopy()
+
+		if !annotations.CompareHashAnnotations(actualVectorConfigMap, desiredVectorConfigMap) {
+			resourceVersion := actualVectorConfigMap.GetResourceVersion()
+			desiredVectorConfigMap.DeepCopyInto(actualVectorConfigMap)
+			actualVectorConfigMap.SetResourceVersion(resourceVersion)
+			if err := remoteClient.Update(ctx, actualVectorConfigMap); err != nil {
+				dcLogger.Error(err, "Failed to update Stargate Vector Agent ConfigMap resource")
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
+		}
+	} else {
+		if err := deleteConfigMapIfExists(ctx, remoteClient, configMapKey, dcLogger); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	dcLogger.Info("Stargate Vector Agent ConfigMap successfully reconciled")
```

Comment:
Nit: this will print this message even if Vector is disabled and there is no config map to reconcile.

Reply:
Yes, that was a debug message I was using earlier. I'll get rid of it.